File: sample/src/main/java/com/mikepenz/aboutlibraries/sample/CustomSortActivity.java
package com.mikepenz.aboutlibraries.sample;

import android.os.Bundle;
import com.mikepenz.aboutlibraries.LibsBuilder;
import com.mikepenz.aboutlibraries.entity.Library;
import com.mikepenz.aboutlibraries.ui.LibsActivity;

import java.io.Serializable;
import java.util.Comparator;


public class CustomSortActivity extends LibsActivity {
    @Override
    public void onCreate(Bundle savedInstanceState) {

        LibsBuilder builder = new LibsBuilder()
            .withFields(R.string.class.getFields())
            .withLibraries("crouton, actionbarsherlock", "showcaseview")
            .withActivityTheme(R.style.MaterialDrawerTheme)
            .withLibraryComparator(new LibraryComparator());

        setIntent(builder.intent(this));
        super.onCreate(savedInstanceState);
    }


    private static class LibraryComparator implements Comparator<Library>, Serializable {

        @Override
        public int compare(Library lhs, Library rhs) {
            // Just to show you can sort however you might want to...
            int result = lhs.getAuthor().compareTo(rhs.getAuthor());
            if (result == 0) {
                // Backwards sort by lib name.
                result = rhs.getLibraryName().compareTo(lhs.getLibraryName());
            }
            return result;
        }
    }
}

File: app/src/main/java/com/mikepenz/aboutlibraries/sample/CustomSortActivity.kt
package com.mikepenz.aboutlibraries.sample

import android.os.Bundle
import com.mikepenz.aboutlibraries.LibsBuilder
import com.mikepenz.aboutlibraries.entity.Library
import com.mikepenz.aboutlibraries.ui.LibsActivity
import java.io.Serializable
import java.util.*


class CustomSortActivity : LibsActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {

        val builder = LibsBuilder()
                .withLibraries("crouton, actionbarsherlock", "showcaseview")
                .withActivityTheme(R.style.MaterialDrawerTheme)
                .withLibraryComparator(LibraryComparator())

        intent = builder.intent(this)
        super.onCreate(savedInstanceState)
    }


    private class LibraryComparator : Comparator<Library>, Serializable {

        override fun compare(lhs: Library, rhs: Library): Int {
            // Just to show you can sort however you might want to...
            var result = lhs.author.compareTo(rhs.author)
            if (result == 0) {
                // Backwards sort by lib name.
                result = rhs.libraryName.compareTo(lhs.libraryName)
            }
            return result
        }
    }
}
